### PR TITLE
Support absolute value tagging in Tensor Mechanics actions

### LINKED ISSUE #24246: Support absolute value tagging in TensorMechanics Master action
## Reason
Testing has shown that the absolute value tagging system (Implemented to address #23399) gives a reliable reference vector for relative convergence. We need support for this to make it easy to use for solid mechanics problems through the TensorMechanics action. This technique is particularly helpful for free expansion problems, where the only BCs are those required to prevent rigid body motion, so the sum of the reactions ends up being zero, which isn't a great reference value. This also needs to be supported for generalized plane strain problems.

## Design
Add the `absolute_value_vector_tags` parameter to `TensorMechanicsAction` and `GeneralizedPlaneStrainAction`, and pass those on the appropriate kernels. Also, some work needs to be done on the `GeneralizedPlaneStrain`. It is already putting the absolute value in the reference vector when `extra_vector_tags` is passed in as a parameter, and we need it to do the same thing when `absolute_value_vector_tags` is passed in. We don't simply want to accumulate the absolute value of the residual, because that accumulation of the absolute value needs to be done in the `GeneralizedPlaneStrainUserObject`.

## Impact
This should only add new capability.

### PR DESCRIPTION
**Pull request by bwspenc**
closes #24246

### CODE REVIEW

**Review by loganharbour — COMMENTED** [review]
Can we get this working without changing the underlying `_xxx_tags` variables? 

I would really prefer that we make those private

**Review by lindsayad — CHANGES_REQUESTED** [review]
Yea we're going to make those data members private in #24206 so this will not work

**Comment by lindsayad** [comment]
Man I was really hoping to lock down the getters for the tags too, but the design of `GeneralizedPlaneStrain` indicates that ship has already sailed

**Comment by lindsayad** [comment]
Mmm I think I can still keep this encapsulated in the framework

**Comment by lindsayad** [comment]
Ok see https://github.com/idaholab/moose/pull/24206/commits/a964f3b697d2b43199a9c9c964871b841e4258b5

**Comment by loganharbour** [comment]
I talked with @bwspenc quite a while about this too. I don't think that the interface knowing about a problem is a good idea

**Comment by lindsayad** [comment]
You don't think that `TaggingInterface` querying information on `ReferenceResidualProblem` is a good idea? We have already installed infrastructure in `TaggingInterface` that is specific to `ReferenceResidualProblem`

**Comment by bwspenc** [comment]
Note that I rebased this against the branch that @tophmatthews and @lindsayad have been working on because this relies on the API changes there to TaggingInterface. We'll want to hold off on merging this until that one gets merged. I did this to verify that the API changes there would meet my needs, which they appear to do.

**Comment by loganharbour** [comment]
> Note that I rebased this against the branch that @tophmatthews and @lindsayad have been working on because this relies on the API changes there to TaggingInterface. We'll want to hold off on merging this until that one gets merged. I did this to verify that the API changes there would meet my needs, which they appear to do.

Great. I tagged you in the other PR, in case you didn't see it. I'm glad to hear it. I'll resolve that comment.

**Comment by tophmatthews** [comment]
@bwspenc can you rebase now that the !24206 is merged? I'd like to test this out as well!

**Comment by bwspenc** [comment]
> @bwspenc can you rebase now that the #24206 is merged? I'd like to test this out as well!

Yeah, just rebased. We'll see how the tests do. I didn't actually run them locally yet, but I assume they'll pass.

**Comment by bwspenc** [comment]
@lindsayad @loganharbour This should be ready to go. It's showing that there are requested changes by @lindsayad, but those have been resolved now. If you don't mind giving this another review, that would be great.

File: modules/tensor_mechanics/src/actions/TensorMechanicsAction.C
Comment (lindsayad) [inline modules/tensor_mechanics/src/actions/TensorMechanicsAction.C:369]:
```suggestion
```

File: modules/tensor_mechanics/test/tests/free_expansion_abs_ref/free_expansion_abs_ref.i
Comment (lindsayad) [inline modules/tensor_mechanics/test/tests/free_expansion_abs_ref/free_expansion_abs_ref.i:5]: Is this necessary?

File: modules/tensor_mechanics/test/tests/free_expansion_abs_ref/tests
Comment (lindsayad) [inline modules/tensor_mechanics/test/tests/free_expansion_abs_ref/tests:7]:
```suggestion
    requirement = 'The tensor mechanics action shall set up the mechanics kernels with the appropriate options to accumulate absolute values of the residual to a tag vector usable for relative convergence checks.'
```

File: modules/tensor_mechanics/test/tests/generalized_plane_strain/tests
Comment (lindsayad) [inline modules/tensor_mechanics/test/tests/generalized_plane_strain/tests:32]:
```suggestion
    requirement = 'The system shall support a generalized plane strain mechanics solution using the reference residual approach to check solution convergence of the field and scalar variables when the reference vector is constructed using absolute value vector tags.'
```

File: modules/tensor_mechanics/test/tests/generalized_plane_strain/tests
Comment (lindsayad) [inline modules/tensor_mechanics/test/tests/generalized_plane_strain/tests:42]:
```suggestion
    requirement = 'The system shall support a generalized plane strain mechanics solution using the reference residual approach to check solution convergence of the field and scalar variables when the reference vector is constructed using unbalanced kernel contributions.'
```

File: modules/tensor_mechanics/test/tests/generalized_plane_strain/tests
Comment (lindsayad) [inline modules/tensor_mechanics/test/tests/generalized_plane_strain/tests:27]:
```suggestion
  [generalized_plane_strain_ref_resid_absolute_value_vector_tags]
```

File: modules/tensor_mechanics/test/tests/generalized_plane_strain/tests
Comment (lindsayad) [inline modules/tensor_mechanics/test/tests/generalized_plane_strain/tests:35]:
```suggestion
  []
  [generalized_plane_strain_ref_resid_extra_vector_tags]
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Review by loganharbour — CHANGES_REQUESTED** [review]
Nothing else from me.

File: modules/tensor_mechanics/test/tests/free_expansion_abs_ref/free_expansion_abs_ref.i
Comment (bwspenc) [reply modules/tensor_mechanics/test/tests/free_expansion_abs_ref/free_expansion_abs_ref.i]: It doesn't appear to be. I removed it.

**Review by loganharbour — APPROVED** [review]

**Review by lindsayad — APPROVED** [review]

File: modules/tensor_mechanics/src/actions/TensorMechanicsAction.C
Comment (dschwen) [inline modules/tensor_mechanics/src/actions/TensorMechanicsAction.C:368]: This could have been a one-liner with `applySpecificParameters`.

**Pull request merged**
